### PR TITLE
[Feature] App drawer support within side navigation

### PR DESCRIPTION
**Pull request by patryk0605**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/mobi/goponents/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: https://github.com/mobi/goponents/issues/60

## What is the new behavior?

This PR introduces an app-drawer referencing the design here: https://tangoe.invisionapp.com/share/JCT2JAKXTQ9#/screens/374819807_Cloud_-_Expanded

It's not a 1:1 match yet, but this PR provides all current expected functionality. Changes to design will be discussed with UX team.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

N/A

### CODE REVIEW

File: projects/go-lib/src/lib/components/go-side-nav/go-side-nav/go-side-nav.component.scss [hunk 1]
```diff
@@ -28,3 +27,39 @@
     width: $side-nav-width--collapsed;
   }
 }
+
+.go-side-nav__nav-items {
+  overflow-y: auto;
```
Comment (grahamhency) [inline projects/go-lib/src/lib/components/go-side-nav/go-side-nav/go-side-nav.component.scss]: Can we add `flex: 1;` to this.

File: projects/go-lib/src/lib/components/go-side-nav/go-side-nav/go-side-nav.component.scss [hunk 2]
```diff
@@ -28,3 +27,39 @@
...
+}
+
+.go-side-nav__app {
+  margin-top: auto;
```
Comment (grahamhency) [inline projects/go-lib/src/lib/components/go-side-nav/go-side-nav/go-side-nav.component.scss]: Can we remove this in favor of the `flex: 1;` being added to the `go-side-nav__nav-items`.

File: projects/go-lib/src/lib/components/go-side-nav/go-side-nav/go-side-nav.component.scss [hunk 3]
```diff
@@ -28,3 +27,39 @@
...
+.go-side-nav__app-icon {
+  display: flex;
+  padding: 0 1rem 0 .5rem;
+}
+
+.go-side-nav__app-drawer {
+  cursor: pointer;
+  margin-left: auto;
+
+  ::ng-deep .go-side-nav__app-drawer--icon {
```
Comment (grahamhency) [inline projects/go-lib/src/lib/components/go-side-nav/go-side-nav/go-side-nav.component.scss]: Don't love this.. but: shrug:

File: projects/go-lib/src/lib/components/go-side-nav/go-side-nav/go-side-nav.component.scss (hunk 1 above)
Comment (patryk0605) [reply projects/go-lib/src/lib/components/go-side-nav/go-side-nav/go-side-nav.component.scss]: added

File: projects/go-lib/src/lib/components/go-side-nav/go-side-nav/go-side-nav.component.scss (hunk 2 above)
Comment (patryk0605) [reply projects/go-lib/src/lib/components/go-side-nav/go-side-nav/go-side-nav.component.scss]: yes, removed

File: projects/go-lib/src/lib/components/go-side-nav/go-side-nav/go-side-nav.component.scss (hunk 3 above)
Comment (patryk0605) [reply projects/go-lib/src/lib/components/go-side-nav/go-side-nav/go-side-nav.component.scss]: yeah, I didn't know a better way of handling this 😞

**Comment by patryk0605** [comment]
@grahamhency addressed your comments, PTAL

**Pull request merged**